vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -186,8 +198,79 @@ def _rpc_call(self, function: str, *args, **kwargs) -> Any:
         self._outgoing.put(request)
         return future.result()
 
-    def process_frame(self, *args, **kwargs) -> DETECTION_NODE_TYPE:
-        return self._rpc_call("process_frame", *args, **kwargs)
+    def process_frame(self,
+                      frame: np.ndarray,
+                      detection_node: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) \
+            -> DETECTION_NODE_TYPE:
+        """Sends a frame to be processed by a worker process.
+        The detection_node input is carefully kept track of, to see if the
+        worker process made any modifications to it.
+        """
+        # Assign individual detection nodes with a _object_id in case they
+        # are modified in the other process, so that they can then be updated
+        input_nodes_by_id: Dict[int, DetectionNode] = {}
```

Comment:
All I need is the id() of the object, not any fancy types. It's purely for matching objects when they come back from the other process.

Reply:
I think it's probably best to use id() too, since it's I think the only thing that changes when pickling/unpickling an object

Reply:
You'd use it like
```python
IdType = NewType('IdType', int)
...
node._object_id = NewType(id(node))
input_nodes_by_id[node._object_id] = node
```

This would prevent someone from accidentally shoving regular ints into the dict without being explicit that they're of type `IdType`

But yeah, it's probably overkill